Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialPropertyConnectionSerializer.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Atom/RPI.Edit/Material/MaterialPropertyConnectionSerializer.h>
+#include <Atom/RPI.Edit/Material/MaterialUtils.h>
+
+#include <AzCore/Serialization/Json/BaseJsonSerializer.h>
+#include <AzCore/Serialization/Json/JsonSerializationResult.h>
+#include <AzCore/Serialization/Json/JsonSerialization.h>
+#include <AzCore/Serialization/Json/StackedString.h>
+
+namespace AZ
+{
+    namespace RPI
+    {
+        namespace JsonMaterialPropertyConnectionSerializerInternal
+        {
+            namespace Field
+            {
+                static constexpr const char type[] = "type";
+                static constexpr const char name[] = "name";
+                static constexpr const char id[] = "id"; // For backward compatibility
+                static constexpr const char shaderIndex[] = "shaderIndex";
```

Comment:
A while back we found out that `static constexpr const char*` is slightly more preferable. As it turned out both the `[]` and `*` variant can return string lengths at compile time, which was believed to be the benefit over `[]`. However the `[]` prevents the string from being added to the global string table and forces a new copy, while the `*` allows the string table to be used. Overall `*` will result in smaller binaries, especially for common words like these.